coderd/updatecheck/updatecheck.go

```diff
@@ -0,0 +1,229 @@
+// Package updatecheck provides a mechanism for periodically checking
+// for updates to Coder.
+//
+// The update check is performed by querying the GitHub API for the
+// latest release of Coder.
+package updatecheck
```

Comment:
We have something similar for the CLI right? Could we use a single package for both instead of having two implementations if so?

Reply:
AFAIK the CLI only checks version differences between API and client, it doesn't check for new GitHub updates. Please correct me if I'm wrong.